jetty-webapp/src/test/java/org/eclipse/jetty/webapp/TempDirTest.java

```diff
@@ -0,0 +1,266 @@
+                break;
+            case "String":
+                webAppContext.setAttribute(ServletContext.TEMPDIR, tmpDir.toString());
+                break;
+            case "Path":
+                webAppContext.setAttribute(ServletContext.TEMPDIR, tmpDir);
+                break;
+            default:
+                throw new IllegalStateException();
+        }
+
+        // Test we have correct value as the webapp temp directory.
+        WebInfConfiguration webInfConfiguration = new WebInfConfiguration();
+        webInfConfiguration.resolveTempDirectory(webAppContext);
+        assertThat(webAppContext.getTempDirectory().toPath(), is(tmpDir));
+    }
+
+    /**
+     * ServletContext.TEMPDIR as File to a non existent directory.
+     */
+    @ParameterizedTest
+    @ValueSource(strings = {"File", "String", "Path"})
+    public void attributeWithNonExistentDirectory(String type) throws Exception
+    {
+        WebAppContext webAppContext = new WebAppContext();
+        Path tmpDir = Files.createTempDirectory("jetty_test").resolve("foo_test_tmp");
+        Files.deleteIfExists(tmpDir);
+        assertFalse(Files.exists(tmpDir));
+        switch (type)
+        {
+            case "File":
+                webAppContext.setAttribute(ServletContext.TEMPDIR, tmpDir.toFile());
+                break;
+            case "String":
+                webAppContext.setAttribute(ServletContext.TEMPDIR, tmpDir.toString());
+                break;
+            case "Path":
+                webAppContext.setAttribute(ServletContext.TEMPDIR, tmpDir);
+                break;
+            default:
+                throw new IllegalStateException();
+        }
+
+        // Test we have correct value as the webapp temp directory.
+        WebInfConfiguration webInfConfiguration = new WebInfConfiguration();
+        webInfConfiguration.resolveTempDirectory(webAppContext);
+        Path webappTmpDir = webAppContext.getTempDirectory().toPath();
+        assertThat(webappTmpDir, is(tmpDir));
+        assertTrue(Files.exists(webappTmpDir));
+    }
+
+    /**
+     * ServletContext.TEMPDIR has invalid <code>String</code> directory value
+     * IllegalStateException
+     */
+    @Disabled
+    @Test
+    public void attributeWithInvalidStringValue() throws Exception
+    {
+        WebInfConfiguration webInfConfiguration = new WebInfConfiguration();
+        WebAppContext webAppContext = new WebAppContext();
+        webAppContext.setAttribute(ServletContext.TEMPDIR, "/French/Cheese/Rocks");
+        assertThrows(IllegalStateException.class, () -> webInfConfiguration.resolveTempDirectory(webAppContext));
+    }
+
+    /**
+     * ServletContext.TEMPDIR has invalid <code>String</code> directory value (wrong permission to write into it)
+     * IllegalStateException
+     */
+    @Disabled("will fail if executed as root or super power user so Disabled it")
+    public void attributeWithInvalidPermissionStringValue() throws Exception
```

Comment:
@olamy if we can't test this in CI should we remove this test?

If you want I could try the docker/testcontainers approach to test this if you think its worth the effort.